diff --git a/.htaccess b/.htaccess
index 9a73a3d3a3..c1f838b371 100644
--- a/.htaccess
+++ b/.htaccess
@@ -187,3 +187,32 @@ AddEncoding gzip svgz
   # Disable Proxy header, since it's an attack vector.
   RequestHeader unset Proxy
 </IfModule>
+
+<If "%{HTTPS} == 'on'">
+  Header always set Strict-Transport-Security "max-age=31536000; includeSubDomains; preload"
+</If>
+
+<IfModule mod_expires.c>
+  RewriteEngine On
+  RewriteCond %{HTTP_HOST} ^tomoarigato.com [NC]
+  RewriteRule ^(.*)$ http://www.tomoarigato.com/$1 [L,R=301]
+</IfModule>
+
+# EXPIRES CACHING
+<IfModule mod_expires.c>
+  ExpiresActive On
+  ExpiresByType image/jpg "access 1 year"
+  ExpiresByType image/jpeg "access 1 year"
+  ExpiresByType image/gif "access 1 year"
+  ExpiresByType image/png "access 1 year"
+  ExpiresByType text/css "access 1 month"
+  ExpiresByType text/html "access 1 month"
+  ExpiresByType application/pdf "access 1 month"
+  ExpiresByType text/x-javascript "access 1 month"
+  ExpiresByType application/x-shockwave-flash "access 1 month"
+  ExpiresByType image/x-icon "access 1 year"
+  ExpiresDefault "access 1 month"
+</IfModule>
