diff --git a/.htaccess b/.htaccess
index b1ee36b..e830603 100644
--- a/.htaccess
+++ b/.htaccess
@@ -98,6 +98,15 @@ AddEncoding gzip svgz
   # RewriteCond %{HTTP_HOST} .
   # RewriteCond %{HTTP_HOST} !^www\. [NC]
   # RewriteRule ^ http%{ENV:protossl}://www.%{HTTP_HOST}%{REQUEST_URI} [L,R=301]
+
+  # http -> https keeping same domain
+  RewriteCond %{ENV:HTTPS} !on
+  RewriteRule ^ https://%{HTTP_HOST}%{REQUEST_URI} [R=301,NE,L]
+
+  # then add www while keeping it https
+  RewriteCond %{HTTP_HOST} !^dev\. [NC]
+  RewriteRule ^ https://dev.%{HTTP_HOST}%{REQUEST_URI} [R=301,NE,L]
+
   #
   # To redirect all users to access the site WITHOUT the 'www.' prefix,
   # (http://www.example.com/foo will be redirected to http://example.com/foo)
@@ -180,3 +189,23 @@ AddEncoding gzip svgz
   # Disable Proxy header, since it's an attack vector.
   RequestHeader unset Proxy
 </IfModule>
+
+<If "%{HTTPS} == 'on'">
+  Header always set Strict-Transport-Security "max-age=31536000; includeSubDomains; preload"
+</If>
+
+# EXPIRES CACHING
+<IfModule mod_expires.c>
+  ExpiresActive On
+  ExpiresByType image/jpg "access 1 year"
+  ExpiresByType image/jpeg "access 1 year"
+  ExpiresByType image/gif "access 1 year"
+  ExpiresByType image/png "access 1 year"
+  ExpiresByType text/css "access 1 month"
+  ExpiresByType text/html "access 1 month"
+  ExpiresByType application/pdf "access 1 month"
+  ExpiresByType text/x-javascript "access 1 month"
+  ExpiresByType application/x-shockwave-flash "access 1 month"
+  ExpiresByType image/x-icon "access 1 year"
+  ExpiresDefault "access 1 month"
+</IfModule>
